apps/service_broadcast/lib/broadcast/init.ex

```diff
@@ -1,16 +1,22 @@
 defmodule Broadcast.Init do
+  use Retry
   use Initializer,
     name: __MODULE__,
     supervisor: Broadcast.Stream.Supervisor
 
   def on_start(state) do
-    Broadcast.Stream.Store.get_all!()
-    |> Enum.reject(&is_nil/1)
-    |> Enum.reject(&Broadcast.Stream.Store.done?(&1))
-    |> Enum.each(fn load ->
-      Broadcast.Stream.Supervisor.start_child({Broadcast.Stream, load: load})
-    end)
-
-    {:ok, state}
+    retry with: constant_backoff(100) do
+      Broadcast.Stream.Store.get_all!()
+      |> Enum.reject(&is_nil/1)
+      |> Enum.reject(&Broadcast.Stream.Store.done?(&1))
+      |> Enum.each(fn load ->
+        Broadcast.Stream.Supervisor.start_child({Broadcast.Stream, load: load})
+      end)
+    after
+      _ ->
+        {:ok, state}
+    else
+      _ -> {:stop, "Could not read state from store", state}
```

Comment:
this else should still return an `{:error, reason}` tuple and then the case statement in the `Init` server handle_continue will translate it into the correct genserver return value.

Reply:
Fixed for this case. I did the same type of set up for all the apps. I'll get this one working and then do the rest.